oci/model.py

```diff
@@ -43,8 +43,24 @@ class OciRegistryType(enum.Enum):
     GCR = 'gcr'
     DOCKERHUB = 'dockerhub'
     ARTIFACTORY = 'artifactory'
+    GHCR = 'ghcr'
+    ALIBABA = 'alibaba'
     UNKNOWN = 'unknown'
 
+    @staticmethod
+    def from_image_ref(image_reference: str) -> 'OciRegistryType':
+        if 'pkg.dev' in image_reference:
+            return OciRegistryType.GAR
+        if 'gcr.io' in image_reference:
+            return OciRegistryType.GCR
+        if 'docker.io' in image_reference:
+            return OciRegistryType.DOCKERHUB
+        if 'ghcr.io' in image_reference:
+            return OciRegistryType.GHCR
+        if 'cr.aliyuncs.com' in image_reference:
```

Comment:
A simple `in` check is not sufficient because we copy over images during release and keep old img-ref as part of the new img-ref.
Rather check with `startswith` (+ some trimming?)

Reply:
By using the netloc as @ccwienk suggested, this should also be handled.